File: cpp/gigasecond/gigasecond.cpp
#include "gigasecond.h"
#include <string>
#include <cmath>

namespace gigasecond {

boost::posix_time::ptime advance(boost::posix_time::ptime time) {
    return time + boost::posix_time::seconds(std::pow(10, 9));
}
	
}

File: go/gigasecond/gigasecond.go
// check if a gigasecond has passed
package gigasecond

import "time"

// Given a moment, return the moment that would be after a gigasecond has
// passed.
//
// A gigasecond is 10^9 (1,000,000,000) seconds.
//
func AddGigasecond(t time.Time) time.Time {
	return t.Add(time.Second * 1e9)
}
